lib/segment/src/index/field_index/geo_index.rs

```diff
@@ -587,6 +594,15 @@ impl PayloadFieldIndex for GeoMapIndex {
             return Some(estimation);
         }
 
+        if let Some(geo_polygon) = &condition.geo_polygon {
+            let geo_hashes = polygon_hashes(geo_polygon, GEO_QUERY_MAX_REGION);
+            let mut estimation = self.match_cardinality(&geo_hashes);
+            estimation
+                .primary_clauses
+                .push(PrimaryCondition::Condition(condition.clone()));
+            return Some(estimation);
+        }
```

Comment:
Since geo-polygon has internal exclusion, it should account it in the estimation.

Reply:
polygon_hashes has taken internal exclusion (interiors) in to consideration. Added an additional step to convert a polygon_wrapper and use it for filtering to reduce the unnecessary building of new Polygon everytime in `check_polygon_intersection` fn.

Reply:
Thanks for pointing out. I didn't realize the precision incompatibility. I pushed a version per your suggestion. However, I am not very confident about "subtract interior estimation". Currently, all tests are passed. But for the new test `test_polygon_with_exclusion`, if I increase the value of `GEO_QUERY_MAX_REGION`, the `real_cardinality` may lower than card.min.

Reply:
> But for the new test test_polygon_with_exclusion, if I increase the value of GEO_QUERY_MAX_REGION, the real_cardinality may lower than card.min.

@zzzz-vincent I have not been able to reproduce this case, which `GEO_QUERY_MAX_REGION` did you use?

Reply:
Per offline discussion with @agourlay , I found that I used some very huge `GEO_QUERY_MAX_REGION` value (eg, 500). This may not be a reasonable number used for any real case.